database/boltdb.go

```diff
@@ -108,38 +73,13 @@ func (db *dbProvider) DbClose() error {
 	return db.DB.Close()
 }
 
-func (db *dbProvider) DbTableRebuild(table DbTable) error {
-	tables := []string{table.Name()}
-	if err := db.DbTableInit(tables); err != nil {
-		return fmt.Errorf("dbInit failed %v", err)
-	}
-
-	table.NewTable()
-
-	err := db.DB.View(func(tx *bolt.Tx) error {
-		b := tx.Bucket([]byte(table.Name()))
-
-		err := b.ForEach(func(k, v []byte) error {
-			vr := bytes.NewReader(v)
-
-			val := table.NewElement()
-			if err := gob.NewDecoder(vr).Decode(val); err != nil {
-				return fmt.Errorf("Decode Error: %v %v %v", string(k), string(v), err)
-			}
-			glog.Infof("%v key=%v, value=%v\n", table, string(k), val)
-
-			return table.Add(string(k), val)
-		})
-		return err
-	})
-	return err
-}
-
-func (db *dbProvider) DbTableInit(tables []string) (err error) {
+func (db *dbProvider) DbTablesInit(tables []string) (err error) {
 
 	glog.Infof("dbInit Tables := %v", tables)
-	for i, v := range tables {
-		glog.Infof("table[%v] := %v, %v", i, v, []byte(v))
+	db.Cache = make(map[string]map[string]interface{})
```

Comment:
Oh, interesting, I didn't noticed that, so, maybe there's no need of Cache for this DB layer, @mcastelino  is this good for the networking code? Is it really necesary to keep the map you had? I think we could delegate this caching task to Bolt.